02.1_Solar_System.rb

```diff
@@ -0,0 +1,172 @@
+    if @has_moons
+      output += "#{bullet}has moons!"
+    else
+      output += "#{bullet}doesn't have any moons :("
+    end
+    if @known_life
+      output += "#{bullet}has life!"
+    end
+    return output
+  end
+
+end # End of Class Planet
+
+
+## SETUP METHODS
+def user_prompt
+  print "(or \"add\" or \"exit\"): "
+  user_input = gets.chomp.downcase
+  return user_input
+end
+
+def user_planet_input #no input checks
+  puts "Please answer the following:"
+
+  print "Planet name: "
+  user_name = gets.chomp.capitalize
+
+  print "Please enter its year length (in Earth days): "
+  user_days = gets.chomp
+
+  print "Please enter its age (in millions of Earth years): "
+  user_age = gets.chomp
+
+  print "Please enter the distance from its sun (in AU): "
+  user_distance = gets.chomp
+
+  print "Please enter the mass in relation to Earth: "
+  user_mass = gets.chomp
+
+  print "Does your planet have moons? "
+  user_moons = gets.chomp
+  case user_moons
+  when "yes","y","true"
+    user_moons = true
+  else
+    user_moons = false
+  end
+  print "Does your planet have known life? "
+  user_life = gets.chomp.downcase
+  case user_life
+  when "yes","y","true"
+    user_life = true
+  else
+    user_life = false
+  end
+
+  # return Planet.new(@user_name, @user_days, @user_age, @user_distance, @user_mass, @user_moons, @user_life)
+  return [user_name, user_days, user_age, user_distance, user_mass, user_moons, user_life]
+end
+
+
+## DEFINE PLANETS & TEST
+planet_1 = Planet.new("Mercury", 87.969, 3_000, 0.387, 0.0553, false, false)
+planet_2 = Planet.new("Venus", 224.7, 1_000, 0.723, 0.815, false, false)
+planet_3 = Planet.new("Earth", Planet::EARTH_YEAR, 500, 1, 1, true, true)
+planet_4 = Planet.new("Mars", 687, 1.52, 1_200, 0.107, true, false)
+planet_5 = Planet.new("Jupiter", 4_332.59, 3_900, 5.2, 317.8, true, false)
+planet_6 = Planet.new("Saturn", 10_759, 400, 9.58, 95.2, true, false)
+
+this_system = SolarSystem.new([planet_1, planet_2, planet_3, planet_4, planet_5, planet_6])
+
+## USER INTERACTION
+puts "========================================\n\n"
+puts "WELCOME TO OUR SOLAR SYSTEM\n\n"
+puts "You can read about the following planets:\n"
+puts this_system.planet_list
+puts "\n========================================\n"
+puts "(Enter \"add\" to add a planet yourself)"
+puts "(Enter \"exit\" to quit this program)"
+print "\nSelect a planet you would like to know more about: "
+
+selection = gets.chomp.downcase
+
+valid = false
+
+until selection == "exit"
+  valid = false
+
+  this_system.planets.each_index do |i|
+    while (i + 1).to_s == selection || this_system.planets[i].name.downcase == selection
+      valid = true
+      puts this_system.planets[i].prettify
+      print "\nPick another planet "
+      selection = user_prompt
```

Comment:
small refactor: You might be able to move the line `selection = user_prompt` to the end of the `until` loop to reduce code repetition!